src/GraphQL.ApiTests/GraphQL.MicrosoftDI.approved.txt

```diff
@@ -62,9 +62,9 @@ namespace GraphQL.MicrosoftDI
         public GraphQL.DI.IServiceRegister Register(System.Type serviceType, System.Func<System.IServiceProvider, object> implementationFactory, GraphQL.DI.ServiceLifetime serviceLifetime, bool replace = false) { }
         public GraphQL.DI.IServiceRegister Register(System.Type serviceType, System.Type implementationType, GraphQL.DI.ServiceLifetime serviceLifetime, bool replace = false) { }
         protected override void RegisterDefaultServices() { }
-        public GraphQL.DI.IServiceRegister TryRegister(System.Type serviceType, object implementationInstance) { }
-        public GraphQL.DI.IServiceRegister TryRegister(System.Type serviceType, System.Func<System.IServiceProvider, object> implementationFactory, GraphQL.DI.ServiceLifetime serviceLifetime) { }
-        public GraphQL.DI.IServiceRegister TryRegister(System.Type serviceType, System.Type implementationType, GraphQL.DI.ServiceLifetime serviceLifetime) { }
+        public GraphQL.DI.IServiceRegister TryRegister(System.Type serviceType, object implementationInstance, GraphQL.DI.RegistrationCompareMode mode = 0) { }
+        public GraphQL.DI.IServiceRegister TryRegister(System.Type serviceType, System.Func<System.IServiceProvider, object> implementationFactory, GraphQL.DI.ServiceLifetime serviceLifetime, GraphQL.DI.RegistrationCompareMode mode = 0) { }
+        public GraphQL.DI.IServiceRegister TryRegister(System.Type serviceType, System.Type implementationType, GraphQL.DI.ServiceLifetime serviceLifetime, GraphQL.DI.RegistrationCompareMode mode = 0) { }
```

Comment:
Isn't TryAddEnumerable only compatible with `Type serviceType, Type implementationType` registrations?  I have not tried to be sure, but I don't understand how it could be otherwise.

Reply:
https://github.com/graphql-dotnet/server/issues/705#issuecomment-1005629696

Reply:
I looked at the code.  It appears we can say with certainty that it will not work with factory methods without further changes to the API, as all factories will have an apparent implementation type of `object`.

Reply:
object? Let's consider ` public static IGraphQLBuilder AddErrorInfoProvider<TProvider>(this IGraphQLBuilder builder, Func<IServiceProvider, TProvider> errorInfoProviderFactory)
            where TProvider : class, IErrorInfoProvider`

`ImplementationFactory.GetType().GenericArguments[1]` is some `TProvider`

Reply:
Oh because the last parameter is `out`.  Ok so that will work.